2294 feat(frontend): styling shortcut keys

Kanav-Arora opened the pull request:
Closes #2294 

PFA the screenshot of UI. I have created a new component for command keys. Inputs like "⌘1" will remain same where as "G then P" will be rendered differently.

<img width="1440" alt="Screenshot 2023-11-03 at 9 29 17 AM" src="https://github.com/twentyhq/twenty/assets/62852363/148723cc-2ec3-4315-b0f0-8b5fb37d7b9a">

lucasbordeau commented on front/src/modules/command-menu/components/CommandItemText.tsx:
```diff
@@ -0,0 +1,45 @@
+  padding-right: ${({ theme }) => theme.spacing(1)};
+`;
+
+export const CommandItemText = ({ text }: CommandItemTextProps) => {
```
Let's not parse the text but create custom props for our use case.

Example : 

{ 
  firstKey: string
  secondKey: string
}

And let the component handle the "then" text.

Kanav-Arora replied:
Should I also pass "then" using props 
There might be case where simultaneous key pressing is required like Ctrl + N
So here it requires '+' instead of 'then'

lucasbordeau commented on front/src/modules/command-menu/components/CommandItemText.tsx:
```diff
@@ -0,0 +1,45 @@
+  white-space: nowrap;
+`;
+
+const StyledCommandKeys = styled.div`
```
This could match the design better : https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=14549-47927&mode=design&t=89WsRl8iB25MjXpq-4

Kanav-Arora commented:
Hi @lucasbordeau 
As requested I have completed the changes but for some categories it requires change in the input in `MenuItemCommand.stories.ts` can you please guide me with that.

P.S: I have changed the design according to figma moreover the commands were not aligned with each other I have fixed that.

<img width="1440" alt="Screenshot 2023-11-05 at 11 26 00 AM" src="https://github.com/twentyhq/twenty/assets/62852363/64e5647f-80ea-4aba-aaaa-95d0efaddf9c">

Kanav-Arora commented:
> Hi @lucasbordeau As requested I have completed the changes but for some categories it requires change in the input in `MenuItemCommand.stories.ts` can you please guide me with that.
> 
> P.S: I have changed the design according to figma moreover the commands were not aligned with each other I have fixed that.
> 
> <img alt="Screenshot 2023-11-05 at 11 26 00 AM" width="1440" src="https://user-images.githubusercontent.com/62852363/280509014-64e5647f-80ea-4aba-aaaa-95d0efaddf9c.png">

Hi @lucasbordeau 
Please help with this

Kanav-Arora commented:
> Hi @lucasbordeau As requested I have completed the changes but for some categories it requires change in the input in `MenuItemCommand.stories.ts` can you please guide me with that.
> 
> P.S: I have changed the design according to figma moreover the commands were not aligned with each other I have fixed that.
> 
> <img alt="Screenshot 2023-11-05 at 11 26 00 AM" width="1440" src="https://user-images.githubusercontent.com/62852363/280509014-64e5647f-80ea-4aba-aaaa-95d0efaddf9c.png">

Hi @charlesBochet 
May be Lucas is busy can you please look into this so that I can proceed.

charlesBochet reviewed (approved):
@Kanav-Arora, thank you!

I've investigated and since we are using command menu library we cannot remove the "data-selected" default behavior. I think we should remove this library, it is not that useful actually. If you have time, we could try to completely remove it and just use a state to track the selected item while pressing arrows up and down.

I've also slightly change the API to introduce firstHotKey: string and secondHotkey: string which is more accurate than command: string[] and I've made some minor style changes to match figma

LGTM :)

Kanav-Arora commented:
> @Kanav-Arora, thank you!
> 
> I've investigated and since we are using command menu library we cannot remove the "data-selected" default behavior. I think we should remove this library, it is not that useful actually. If you have time, we could try to completely remove it and just use a state to track the selected item while pressing arrows up and down.
> 
> I've also slightly change the API to introduce firstHotKey: string and secondHotkey: string which is more accurate than command: string[] and I've made some minor style changes to match figma
> 
> LGTM :)

Hi @charlesBochet 
Don't you want the requested changes? Or will it be a seperate issue?

charlesBochet commented:
It should be a separate issue! Would love to get help on it :)

Kanav-Arora commented:
> It should be a separate issue! Would love to get help on it :)

Sure!
